test/api/model/initial_snapshot_test.dart

```diff
@@ -2,7 +2,29 @@ import 'package:checks/checks.dart';
 import 'package:test/scaffolding.dart';
 import 'package:zulip/api/model/initial_snapshot.dart';
 
+import '../../stdlib_checks.dart';
+
 void main() {
+  test('UnreadMessagesSnapshot from json recognizes channels as streams', () {
+    final snapshot = UnreadMessagesSnapshot.fromJson({
+      'count': 1,
+      'pms': <dynamic>[],
+      'huddles': <dynamic>[],
+      'mentions': <dynamic>[],
+      'old_unreads_missing': false,
+      'streams': [{
+        'stream_id': 1,
+        'topic': 'topic name',
+        'unread_message_ids': [1, 2]}]
+    });
+
+    check(snapshot.channels.length).equals(1);
+    check(snapshot.channels).single.jsonEquals(
```

Comment:
What's the reasoning for keeping the separate `length` check here? (Compare my suggestion at https://github.com/zulip/zulip-flutter/pull/839#discussion_r1694042124 .)

Reply:
Oh.. Sorry I missed this one. I've pushed a new revision just now, please take a look.